app/api/agent/engineer/route.ts

```diff
@@ -0,0 +1,37 @@
+import {engineer} from '~/agents/engineer'
+import env from '~/env.mjs'
+
+export const maxDuration = 300
+
+const validateAgentCall = (headers: Headers) => {
+	if (!(headers.get('Authorization') === `Bearer ${env.NEXTAUTH_SECRET}`))
+		throw new Error('Webhook signature could not be verified')
+}
+
+export const POST = async (req: Request) => {
+	console.log('POST /api/agent/engineer')
+
+	validateAgentCall(req.headers)
```

Comment:
Does this need a try-catch wrapper? With `return(500)` in the catch

Reply:
the response is not awaited because it's desynced

Reply:
We do likely need better error handling - but this needs to happen to the DB (to be reflected in UI). Will address this in a future PR with schema changes to Run table